File: Day7/Sort 0  1/Sort0_1.py
def binSort(self, A, N): 
        #Your code here
        #No need to print the array
    
        i = 0
        j = N - 1
        
        while i <= j:
            if A[i] == 1:
                while j > 0 and A[j] == 1:
                    j -= 1
                    
                if i <= j:
                    A[i], A[j] = A[j], A[i]
                    i += 1
                    j -= 1
            else:
                while i < N and A[i] == 0:
                    i += 1

File: Day7/Sort 0  1/Sort0_1.cpp
#include<algorithm>

using namespace std;

class Solution{
  public:
    
    // A[]: input array
    // N: input array
    //Function to sort the binary array.
    void binSort(int A[], int N)
    {
       //Your code here
       
       /**************
        * No need to print the array
        * ************/
        int i = 0, j = N - 1;
        
        while(i <= j)
        {
            if(A[i] == 1)
            {
                while(A[j] == 1)
                {
                    j--;
                }
                
                if(i <= j)
                {
                    swap(A[i], A[j]);
                    i++;
                    j--;
                }
            }
            else
            {
                while(A[i] == 0)
                {
                    i++;
                }
            }
        }
    }
};